webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -397,174 +397,84 @@ object JsonResponseObjects {
-          override val version:       Version,
-          override val fullName:      String,
-          override val servicePack:   Option[String],
-          override val kernelVersion: Version
-      ) extends OsDetails(name, version, fullName, servicePack, kernelVersion)
-
-      final case class Windows(
-          override val name:          String,
-          override val version:       Version,
-          override val fullName:      String,
-          override val servicePack:   Option[String],
-          override val kernelVersion: Version
-      ) extends OsDetails(name, version, fullName, servicePack, kernelVersion)
-
-      // we need a transformer for each subtype because "name" cannot be configured in the class ADT
-      implicit val unknownTransformer: Transformer[domain.UnknownOS, UnknownOS] = Transformer
-        .define[domain.UnknownOS, UnknownOS]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-      implicit val linuxTransformer:   Transformer[domain.Linux, Linux]         = Transformer
-        .define[domain.Linux, Linux]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-      implicit val solarisTransformer: Transformer[domain.Solaris, Solaris]     = Transformer
-        .define[domain.Solaris, Solaris]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-      implicit val bsdTransformer:     Transformer[domain.Bsd, Bsd]             = Transformer
-        .define[domain.Bsd, Bsd]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-      implicit val aixTransformer:     Transformer[domain.Aix, Aix]             = Transformer
-        .define[domain.Aix, Aix]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-      implicit val windowsTransformer: Transformer[domain.Windows, Windows]     = Transformer
-        .define[domain.Windows, Windows]
-        .withFieldComputed(_.name, _.os.name)
-        .buildTransformer
-
-      implicit val transformer: Transformer[domain.OsDetails, OsDetails] = Transformer.derive[domain.OsDetails, OsDetails]
+    /**
+     * Needed to handle the serialization of the "type" of a machine even if it there is no machine.
+     */
+    sealed abstract class MachineType(override val entryName: String) extends EnumEntry {
+      def name: String = entryName
     }
 
-    @jsonDiscriminator("type")
-    sealed abstract class MachineInfo(
-        val id:           Option[MachineUuid],
-        val provider:     Option[VmType],
-        val manufacturer: Option[Manufacturer],
-        val serialNumber: Option[String]
-    )
+    object MachineType extends Enum[MachineType] {
+      case object UnknownMachineType  extends MachineType("Unknown")
+      case object PhysicalMachineType extends MachineType("Physical")
+      case object VirtualMachineType  extends MachineType("Virtual")
+      case object NoMachine           extends MachineType("No machine Inventory")
```

Comment:
I simplified the `MachineInfo` ADT to make it closest to the domain object of the same name i.e. instead of having the type discriminator on an sum type ADT, just map the `machineType` attribute from the case class to this custom `MachineType` one...

(I should have done that from the start :see_no_evil:)